constraintlayout/swing/src/main/java/org/constraintlayout/swing/MotionLayout.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.constraintlayout.swing;
+
+import androidx.constraintlayout.core.motion.utils.Utils;
+import androidx.constraintlayout.core.state.WidgetFrame;
+import androidx.constraintlayout.core.widgets.ConstraintWidget;
+import org.constraintlayout.swing.core.motion.model.MotionEngine;
+
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+
+/**
+ * Basic implementation of ConstraintLayout as a Swing LayoutManager
+ */
+public class MotionLayout implements LayoutManager2 {
+    private static final boolean DEBUG = false;
+    MotionEngine motionEngine = new MotionEngine();
+    Container mParent;
+      final HashMap<Component, String> mViewsToIds = new HashMap<>();
+    private final HashMap<String, Component> mIdsToViews = new HashMap<>();
+
+    public MotionLayout() {
+    }
+
+    public MotionLayout(@Language("JSON5") String content) {
+        parse(content);
+    }
+
+    public MotionLayout(@Language("JSON5") String content, Runnable runnable) {
+        parse(content);
+        SwingUtilities.invokeLater(runnable);
+    }
```

Comment:
Add documentation. It's not clear what's the purpose of the runnable. It's not different from doing the same thing in the caller:
`MotionLayout myMotionLayout = new MotionLayout(content);`
`SwingUtilities.invokeLater(runnable)`